src/main/java/com/pusher/client/channel/impl/PresenceChannelImpl.java

```diff
@@ -187,9 +163,24 @@ private static PresenceData extractPresenceDataFrom(final String message) {
     }
 
     @SuppressWarnings("rawtypes")
-    private void storeMyUserId(final Object channelData) {
-        final Map channelDataMap = GSON.fromJson((String)channelData, Map.class);
-        myUserID = String.valueOf(channelDataMap.get("user_id"));
+    private String extractUserIdFromChannelData(final String channelData) {
+        final Map channelDataMap;
+        try {
+            channelDataMap = GSON.fromJson((String)channelData, Map.class);
+        } catch (final JsonSyntaxException e) {
+            throw new AuthorizationFailureException("Invalid response from Authorizer: unable to parse channel_data object: " + channelData, e);
+        }
+        Object maybeUserId;
+        try {
+            maybeUserId = channelDataMap.get("user_id");
```

Comment:
I don't think `get` throws a null pointer exception if the key doesn't exist, it just returns null, so I don't think the catch block would ever be called. The `if (maybeUserId == null)` check below is sufficient.

In the previous version, `String.valueOf(null)` would have been where the null pointer exception was thrown I think.

Reply:
Or were you catching `NullPointerException` because `channelDataMap` might be null?

Reply:
No, I was checking for `NullPointerException` because the `Map` interface "permits" (via documentation) that implementing classes can either return `null` or throw a `NullPointerException` (annoyingly). Even though the concrete type we are using returns `null`, I don't want to rely on this since at this level we are dealing with a `Map`, and shouldn't have to worry about what the concrete implementation is (which unfortunately means checking both cases...).

I don't feel too strongly about this though (I appreciate it could be considered overly cautious). Do you?

Reply:
Ah I see. I think that sounds OK. Overly cautious is fine with me.